### PR TITLE
Fix formatting in the FX Graph Mode Quantization guide

### LINKED ISSUE #2337: [BUG] - Fix formating
### Add Link

https://pytorch.org/tutorials/prototype/fx_graph_mode_quant_guide

### Describe the bug

Fix formatting in the tutorial: https://github.com/pytorch/tutorials/blob/main/prototype_source/fx_graph_mode_quant_guide.rst#L73) 

Right now only "1.a. Symbolically trace only the code that needs to be quantized" appears due to extraneous ############## lines. 

To do:

1. Remove the extraneous ###### lines
2. Remove numbering from the headings.
3. Make editorial changes as needed.
4. Add line breaks where needed. Example:
   
Screenshot 2023-05-23 at 10 31 17 AM (2)

### Describe your environment

Website

cc @carljparker

### PR DESCRIPTION
**Pull request by fabiogomez11c**
Fixes #2337

## Description
Fix formatting in the tutorial: https://pytorch.org/tutorials/prototype/fx_graph_mode_quant_guide

## Checklist

- [x] Remove the extraneous ###### lines
- [x] Remove numbering from the headings.
- [x] Add line breaks where needed
- [x] Make editorial changes as needed.

cc @svekars @carljparker

### CODE REVIEW

**Comment by svekars** [comment]
Please sign the CLA so we can review your PR. Thank you for contributing to PyTorch!

File: prototype_source/fx_graph_mode_quant_guide.rst [hunk 1]
```diff
@@ -207,8 +205,8 @@ non-traceable logic, wrapped in a module
   class FP32NonTraceable:
       ...
 
-
-2. Define observed version of FP32NonTraceable
+2. Define observed version of
+FP32NonTraceable
```
Comment (mandroid6) [inline prototype_source/fx_graph_mode_quant_guide.rst:209]: Why do we need this on a newline?

**Comment by svekars** [comment]
Thanks so much for contributing. Makes sure the following is addressed: 
1. Please sign the CLA
2. Provide a descriptive title in your PR - Like: Fix formatting in the FX Graph Mode Quantization guide

**Comment by svekars** [comment]
It looks like the code blocks are still not visible in the HTML. Compare the source:
source-gh
and the thml
html-page1

Also, we can remove the numbering from the titles completely:

**Comment by fabiogomez11c** [comment]
ok, I'll do it

**Comment by fabiogomez11c** [comment]
@svekars I have a question, there are some TODOs inside this tutorial about adding some links, do you have in mind to add that in the future?

**Review by svekars — APPROVED** [review]
Looks great! Thanks so much, Fabio!

**Pull request merged**